src/main/java/com/cloudbees/jenkins/github/AccessTokenCredential.java

```diff
@@ -0,0 +1,109 @@
+package com.cloudbees.jenkins.github;
+
+import com.cloudbees.plugins.credentials.CredentialsDescriptor;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AutoCompletionCandidates;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.kohsuke.github.GHAuthorization;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class AccessTokenCredential extends BaseStandardCredentials implements StandardUsernameCredentials {
+
+    private final String githubServer;
+
+    private final String username;
+
+    private final Secret token;
```

Comment:
@ndeloof the token is really a password with restricted permissions. The user gets to select how much permissions to give the token. To my way of thinking a better way would be to have domain specifications and domain requirements that match the permissions that can be granted to a github token. That way users can put the tokens into the domains that match the permissions they granted.

A username/password would have full permissions, while a username/token would be in a different domain. As such I am unclear that the token needs to be a separate type from the username/password. Things that would argue for a separate type:

* The auth with a token being different than the auth with a password, e.g. when using a password you have to pass `Authenication: Basic ...` but with a token you have to pass `Authentication: Token ...`
* You are providing a special API to generate the token (e.g. via a web flow)... if the resulting token behaves just like a password in terms of authentication then you would implement the `StandardUsernamePasswordsCredentials` interface but use a button or such to control generating/validation/updating the token and you would not display a "password" field.
* Perhaps some other exotic use case I cannot think of right now.

Absence the above, I would favour not creating a special type for this.

In any case I think you need to look at creating some `GitHubPermissionDomainRequirement` and `GitHubPermissionDomainSpecification` style class(es) so that users can put tokens into domains that match what the tokens are good for.